app/code/Magento/Bundle/Test/Unit/Pricing/Price/ConfiguredPriceTest.php

```diff
@@ -0,0 +1,185 @@
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Bundle\Test\Unit\Pricing\Price;
+
+use Magento\Bundle\Pricing\Price\DiscountCalculator;
+use Magento\Catalog\Model\Product;
+use Magento\Catalog\Model\Product\Configuration\Item\ItemInterface;
+use Magento\Bundle\Pricing\Price\ConfiguredPrice;
+use Magento\Bundle\Pricing\Adjustment\Calculator;
+use Magento\Catalog\Pricing\Price\ConfiguredPriceSelection;
+use Magento\Framework\DataObject;
+use Magento\Framework\Pricing\Amount\AmountInterface;
+use Magento\Framework\Pricing\Price\PriceInterface;
+use Magento\Framework\Pricing\PriceCurrencyInterface;
+use Magento\Framework\Pricing\PriceInfo\Base;
+use Magento\Framework\Serialize\Serializer\Json;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Test for \Magento\Bundle\Pricing\Price\ConfiguredPrice
+ */
+class ConfiguredPriceTest extends TestCase
+{
+    /**
+     * @var float
+     */
+    private $basePriceValue = 100.;
+
+    /**
+     * @var ItemInterface|MockObject
+     */
+    private $itemMock;
+
+    /**
+     * @var Product|MockObject
+     */
+    private $productMock;
+
+    /**
+     * @var MockObject
+     */
+    private $calculator;
+
+    /**
+     * @var Base|MockObject
+     */
+    private $priceInfoMock;
+
+    /**
+     * @var ConfiguredPrice
+     */
+    private $model;
+
+    /**
+     * @var PriceCurrencyInterface|MockObject
+     */
+    private $priceCurrencyMock;
+
+    /**
+     * @var Json|MockObject
+     */
+    private $jsonSerializerMock;
+
+    /**
+     * @var ConfiguredPriceSelection|MockObject
+     */
+    private $configuredPriceSelectionMock;
+
+    /**
+     * @var AmountInterface|MockObject
+     */
+    private $amountInterfaceMock;
+
+    /**
+     * @var DiscountCalculator|MockObject
+     */
+    private $discountCalculatorMock;
+
+    /**
+     * Initialize base dependencies
+     */
+    protected function setUp(): void
+    {
+        $basePrice = $this->getMockForAbstractClass(PriceInterface::class);
+        $basePrice->expects($this->any())->method('getValue')->willReturn($this->basePriceValue);
+
+        $this->priceInfoMock = $this->createMock(Base::class);
+        $this->priceInfoMock->expects($this->any())->method('getPrice')->willReturn($basePrice);
+        $this->productMock = $this->getMockBuilder(Product::class)
+            ->setMethods(['getPriceInfo', 'getOptionById', 'getResource', 'getId'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->productMock->expects($this->once())->method('getPriceInfo')->willReturn($this->priceInfoMock);
+        $this->productMock->expects($this->any())->method('getId')->willReturn(123);
+
+        $this->itemMock = $this->getMockBuilder(ItemInterface::class)
+            ->getMock();
+        $this->itemMock->expects($this->any())->method('getProduct')->willReturn($this->productMock);
```

Comment:
I believe it would be much better if you will move this part to `testGetValueMethod` method.